### DISCUSSION TITLE
Showing off: Custom V9 components

### DISCUSSION THREAD

**Discussion by blendsdk** [microsoft/fluentui#28314] [Show and tell]
I am working to migrate my v8 components to v9. So far I find the v9 `Slots API` to be very cool and great for customizing. 
I have derived a simplified version of the `Slots API` that works better for my needs. Below is a link to the storybook.

Take a look at the [AppLauncherModule](https://blendsdk.truesoftware.nl/?path=/story/applaunchermodule--initial) and the [AppBarTabList](https://blendsdk.truesoftware.nl/?path=/story/appbartablist--initial)

![image](https://github.com/microsoft/fluentui/assets/5524595/76bbd094-1e0d-46b4-b917-75944f94a0cf)
![image](https://github.com/microsoft/fluentui/assets/5524595/e2da5e25-b8c1-413a-9d78-b4cf9d27ef13)
![image](https://github.com/microsoft/fluentui/assets/5524595/bd35042e-5c75-496f-b3cb-625105d79e8f)

 Let me know what you think :)

**Comment by tombohub** [comment]
Nicely done! And these are using fluentui v9 tokens?

**Reply by blendsdk** [reply]
> Nicely done! And these are using fluentui v9 tokens?

Yes. I have used `tokens` from v9. Everything you there is v9

**Comment by Patchrik** [comment]
@blendsdk 
Did you have to do anything special to get Fluent UI to render the Icons in your components? I'm using Vite with storybook. And I've got everything but icons working, but can't figure out what the issue is.

**Reply by blendsdk** [reply]
Hi @Patchrik, I don't have a specific setup; just the standard config stuff StoryBook generated. Perhaps you could check the generated HTML elements when the component is rendered. FUI9 icons are rendered as standard inline SVG elements. You should get something like this screenshot. Maybe the CSS enclosing the SVG element gets mixed up with something else. Are you using Icons provided by FUI9 or some other library like FontAwesome?

![Screenshot 2025-01-09 at 01 22 14](https://github.com/user-attachments/assets/a6ae651b-5585-489c-a1ca-8d4d60b3db94)
